diff --git a/app/channel.go b/app/channel.go
index 020089eb5..cd04ecae3 100644
--- a/app/channel.go
+++ b/app/channel.go
@@ -337,6 +337,13 @@ func (a *App) GetOrCreateDirectChannel(c *request.Context, userID, otherUserID s
 			}
 		}
 	}
+	// __MATTERFOSS__: To prevent harassement and spam, we do not allow guests to create private chats
+	{
+		user, err := a.GetUser(userID)
+		if err == nil && user.IsGuest() {
+			return nil, model.NewAppError("createDirectChannel", "api.channel.create_channel.direct_channel.guest_restricted_error", nil, "", http.StatusBadRequest)
+		}
+	}
 
 	channel, err := a.createDirectChannel(userID, otherUserID, channelOptions...)
 	if err != nil {
